libeos-payg/efi.c

```diff
@@ -559,6 +559,26 @@ eospayg_efi_secureboot_active (void)
   return !!content[0];
 }
 
+/* eospayg_efi_setupmode_active:
+ *
+ * Check if the system was booted with SetupMode
+ *
+ * Returns: %TRUE if booted with SetupMode, %FALSE otherwise
+ */
+gboolean
+eospayg_efi_setupmode_active (void)
+{
+  g_autofree char *tname = full_efi_name (GLOBAL_VARIABLE_GUID, "SetupMode");
+  g_autofree unsigned char *content = NULL;
+  int size;
+
+  content = efivarfs_read (tname, &size, NULL);
+  if (!content || size != 1)
+    return FALSE;
+
+  return !!content[0];
```

Comment:
This function's body is the same as eospayg_efi_secureboot_active just with a different variable name. Perhaps you could define a generic `eospayg_efi_read_boolean()` function instead.